packages/devtools/src/getMonolithicCSSRules.ts

```diff
@@ -0,0 +1,55 @@
+import * as stylis from 'stylis';
+import type { AtomicRules, MonolithicAtRules, MonolithicRules, RuleDetail } from './types';
+
+function parseRuleElement(monolithicRules: MonolithicRules, element: stylis.Element, overriddenBy?: string) {
+  // example of `value`: `.f3xbvq9:hover`
+  // `children` contains all css under the `value` selector
+  const { value: classNameSelector, children } = element;
+
+  const nestedSelector = stylis.tokenize(classNameSelector).slice(1).join('');
+  monolithicRules[nestedSelector] = monolithicRules[nestedSelector] ?? [];
+
+  (children as stylis.Element[]).forEach(child => {
+    (monolithicRules[nestedSelector] as RuleDetail[]).push({
+      css: child.value,
+      className: stylis.tokenize(classNameSelector)[0].slice(1),
+      overriddenBy,
+    });
+  });
```

Comment:
```suggestion
  if (Array.isArray(children)) {
    children.forEach(child => {
      (monolithicRules[nestedSelector] as RuleDetail[]).push({
        css: child.value,
        className: stylis.tokenize(classNameSelector)[0].slice(1),
        overriddenBy,
      });
    });
  } else {
    throw new Error('Unsupported input from "element.children"');
  }
```

I would use an explicit check to avoid typecasts